Closes #234 Add new ADRS vignette (#240)
pharmaverse/admiralonco: vignettes/adrs.Rmd

)
```

> ### Flag Worst Assessment at Each Date (`ANL01FL`) {#anl01fl}

Author: Yes, updated

  )
```

> ### Flag Assessments up to First PD (`ANL02FL`) {#anl02fl}

Reviewer: Shall we update this in adrs_basic too? as your new title is more descriptive

Author: Yes, updated

  )
```

```{r, echo=FALSE}
dataset_vignette(
  adrs,
  display_vars = exprs(USUBJID, AVISIT, PARAMCD, AVALC, ADT, ANL01FL, ANL02FL)
)
```

### Select Source Assessments for Parameter derivations

For most parameter derivations the post-baseline overall response assessments up
to and including first PD are considered.
```{r}
ovr <- filter(adrs, PARAMCD == "OVR" & ANL01FL == "Y" & ANL02FL == "Y")
```

```{r, echo=FALSE}

> dataset_vignette(

Reviewer: the 3 patients shown here aren't great examples in the rendered vignette as all their first assessments were `PD` so they just get those records kept, maybe choose a better example patient to include to show this?

Author: Should be solved by the new test data (https://github.com/pharmaverse/admiralonco/issues/226).

Reviewer: ok @bundfussr  will trust you to double check this later

Author: Meanwhile, the new data is used, i.e., it should look better now.

  ovr,
  display_vars = exprs(USUBJID, AVISIT, AVALC, ADT, RANDDT)
)
```

## Define Events

> Common events like complete response (CR), partial response (PR), ... are

Reviewer: ```suggestion
The building blocks for the events that contribute to deriving common endpoints like what constitutes a responder, or a Best Overall Response of complete response (CR), ... are
```
I feel like this would read better

Author: Updated

predefined in admiralonco (see [Pre-Defined Event
Objects](../reference/event_objects.html)). Some may need to be adjusted for
study-specific needs, e.g., minimum time between response and confirmation

> assessment. Here the `keep_source_vars` argument is updated.

Reviewer: is there a better example we could use here? as this change is very subtle and most users won't easily understand the difference. how about a hypothetical example where there was another possible response value in your study and then you show how to create a new version of `rsp_y` to include this along with `CR` and `PR`? that might then already help people to visualise the iRECIST examples to come where `iCR` and `iPR` could be used.

Author: I have added an example for the confirmation period as this is already mentioned in the text.

  )
```

```{r, echo=FALSE}
dataset_vignette(
  adrs,
  display_vars = exprs(USUBJID, AVISIT, PARAMCD, AVALC, ADT, ANL01FL, ANL02FL)
)
```

### Select Source Assessments for Parameter derivations

For most parameter derivations the post-baseline overall response assessments up
to and including first PD are considered.
```{r}
ovr <- filter(adrs, PARAMCD == "OVR" & ANL01FL == "Y" & ANL02FL == "Y")
```

```{r, echo=FALSE}
dataset_vignette(
  ovr,
  display_vars = exprs(USUBJID, AVISIT, AVALC, ADT, RANDDT)
)
```

## Define Events

The building blocks for the events that contribute to deriving common endpoints
like what constitutes a responder, or a Best Overall Response of complete

> response (CR), ... are predefined in admiralonco (see [Pre-Defined Event

Reviewer: ```suggestion
response (CR), ... are predefined in admiralonco (see [Pre-Defined Response Event
```

Author: Updated

Objects](../reference/event_objects.html)). Some may need to be adjusted for
study-specific needs, e.g., minimum time between response and confirmation

> assessment. Here the confirmation period ad the `keep_source_vars` argument is

Reviewer: ```suggestion
assessment. Here the confirmation period and the `keep_source_vars` argument is
```
typo fix?

Author: Fixed

they could equally be achieved via `ADSL` dates or `ADEVENT` parameters.
If you prefer to store as an ADSL date, then the function
`admiral::derive_var_extreme_dt()` could be used to find the date of
first `PD` as a variable, rather than as a new parameter record.

## Derive Response Parameter {#rsp}

> The function `derive_extreme_event()` can then be used to find the date of first

Reviewer: ```suggestion
The function `admiral::derive_extreme_event()` can then be used to find the date of first
```

Reviewer: same comment in any other places, won't repeat this

Author: Updated

response. In the below example, the response condition has been defined as `CR`

> or `PR` via the `rsp_y`[^1] event.

Reviewer: no comment here: only wanted to say that the `[^1]` approach here is very very cool! i like this a lot in how it renders in the vignette and might be worth sharing this at the admiral core dev team sometime as maybe others vignettes could benefit from this. was a new learning for me!

Author: It is already used in other vignettes, e.g., the questionnaires vignette. However, most likely not all developers are aware of this feature. I have created an issue to add it to the "Writing Vignette" vignette (https://github.com/pharmaverse/admiraldev/issues/324).

## Derive Clinical Benefit Parameter {#cb}

The function `derive_extreme_event()` can then be used to derive the clinical
benefit parameter, which we define as a patient having had a response or a
sustained period of time before first `PD`. This could also be known as disease
control. In this example the "sustained period" has been defined as 42 days
after randomization date via the `cb_y`[^2] event.

[^2]:
```{r}
cb_y
```
```{r}
adrs <- adrs %>%
  derive_extreme_event(
    by_vars = exprs(STUDYID, USUBJID),
    order = exprs(desc(AVALC), ADT, PARAMCD),
    mode = "first",
    events = list(

>       event(

Reviewer: why is this not using `rsp_y` directly?

Reviewer: for me it fits closer with your new style of building these derivations up with the events passed as source objects so i would say worth to update for a cleaner and more consistent story

Author: OK, I have updated it.

        condition = PARAMCD == "RSP" & AVALC == "Y",
        set_values_to = exprs(AVALC = "Y")
      ),
      cb_y,
      no_data_n
    ),
    source_datasets = list(
      ovr = ovr,
      adsl = adsl
    ),

>     ignore_event_order = TRUE,

Reviewer: could we add an explanation of why this is set `TRUE` here whereas it wasn't in the above derived parameter? reading the admiral PR i'm trying to make sense of this argument and my best guess is that we want the earliest of either the response or the sustained benefit (i.e. it doesn't matter which), whereas in BOR it would matter as we want the best first. then the `no_data_n` i presume gets always checked last anyway as this is a non-event with `AVALC` being set as `"N"`. by saying the event order is ignored is confusing to me as the `no_data_n` could be given equal weight then.

they could equally be achieved via `ADSL` dates or `ADEVENT` parameters.
If you prefer to store as an ADSL date, then the function
`admiral::derive_var_extreme_dt()` could be used to find the date of
first `PD` as a variable, rather than as a new parameter record.

## Derive Response Parameter {#rsp}

> The function `adiral::derive_extreme_event()` can then be used to find the date

Reviewer: ```suggestion
The function `admiral::derive_extreme_event()` can then be used to find the date
```

Reviewer: this typo exists in few places - please fix all

Author: All fixed

    dataset_ref = adsl,
    dataset_add = adrs,
    by_vars = exprs(STUDYID, USUBJID),

>     filter_add = PARAMCD == "BOR" & AVALC %in% c("CR", "PR"),

Reviewer: same update needed for adrs_basic?

Author: It does not affect the results. It just makes the code a bit shorter. I have updated adrs_basic and the templates.

    )
  )

adrs <- adrs %>%
  derive_extreme_event(
    by_vars = exprs(STUDYID, USUBJID),
    order = exprs(desc(AVALC), ADT, PARAMCD),
    mode = "first",
    events = list(

>       event(

Reviewer: same comment as above - why not create a `crsp_y` for this?

Author: See my comment above.

    dataset_ref = adsl,
    dataset_add = adrs,
    by_vars = exprs(STUDYID, USUBJID),

>     filter_add = PARAMCD == "CBOR" & AVALC %in% c("CR", "PR"),

Reviewer: same updates needed in adrs_basic?

Author: Updated

)
```

## Derive Non-standard Parameters {#nonstandard}

As `admiral::derive_extreme_event()` is very flexible, it is easy to implement
non-standard parameters. Below two examples for modified RECIST 1.1 parameters
are shown.

### Adding a Criterion for Confirmed Clinical Benefit
Confirmed clinical benefit was defined before as confirmed response or CR, PR,
SD, or NON-CR/NON-PD at least 42 days after randomization. Here an alternative
definition is implemented which considers PD more than 42 days after
randomization as an additional criterion for clinical benefit.
```{r}
adrs <- adrs %>%
  derive_extreme_event(
    by_vars = exprs(STUDYID, USUBJID),
    order = exprs(desc(AVALC), ADT, PARAMCD),
    mode = "first",
    events = list(
      event(
        condition = PARAMCD == "CRSP" & AVALC == "Y",
        set_values_to = exprs(AVALC = "Y")
      ),
      cb_y,

>       event(

Reviewer: good example! why don't we create this as an object first in the example and then feed it in to the function call? might be easier for users to follow and then its more consistent to how you build the other parameter derivations

Author: Updated

    order = exprs(ADT),
    mode = "first",
    source_datasets = list(
      ovr = ovr,
      adsl = adsl
    ),
    events = list(bor_cr, bor_pr, bor_sd, bor_non_crpd, bor_ned, bor_pd, bor_ne, no_data_missing),
    set_values_to = exprs(
      PARAMCD = "A1BOR",
      PARAM = paste(
        "Best Overall Response by Investigator (confirmation not required)",
        "- Recist 1.1 adjusted for NED at Baseline"
      ),
      PARCAT1 = "Tumor Response",
      PARCAT2 = "Investigator",
      PARCAT3 = "Recist 1.1 adjusted for NED at Baseline",
      AVAL = aval_resp(AVALC),
      ANL01FL = "Y"
    )
  )
```

## Derive Parameters using Independent Review Facility (IRF)/Blinded Independent Central Review (BICR) responses {#irf}

All of the above steps can be repeated for different sets of records, such as
now using assessments from the IRF/BICR instead of investigator. For this you
would just need to replace the first steps with selecting the required records,
create the variables `AVALC`, `AVAL`, `ADT`, `AVISIT`, `ANL01FL`, `ANL02FL` and
the dataset `ovrb` (see [Pre-processing of Input Records](#input)) and then feed
these as input to the downstream parameter functions.

```{r}

> adrs_bicr <- rs %>%

Reviewer: if updating this best to also update in adrs_basic and templates for consistency

Author: I have updated adrs_basic. In the templates we do not have this code.
